### ISSUE TITLE
[Spike] Get rid of Gitea

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#308]
Right now Gitea provides the following functionality:

- Orgs concept (they are implemented by querying git)
- Storage for the application code
- Triggering hook for Tekton
- Pushing mechanism (through `git push`)

## Why remove it then?

- Many responsibilities (no separation of concerns)
- Hackish pushing method (shell script?!)
- Triggering hook is a weak link. When it's not accessible, Tekton doesn't trigger and the user has no idea why
- Git may be slow with applications that have big assets. This is because git is not good with large files and repos.
- Pushing applications through the web UI is not possible if we only support git.

**Comment by agracey** [comment]
I've also had some issues where a delete then push breaks the hook

**Comment by jimmykarily** [comment]
@agracey sent this as a possible alternative to minio: https://github.com/chrislusf/seaweedfs

**Comment by jimmykarily** [comment]
We already started creating separate stories to implement the following concept:

- Gitea stays but only as a storage for code and not an integral part of the push process
- We create an upload API endpoint and we use that to upload the code as a tarball #308 #338 
- Eventually we support other ways to provide the sources (e.g. directly pull from GitHub)
- We can use the web UI to upload the code and push the app.

There are or will be new stories about each one of these. Closing this Spike.
